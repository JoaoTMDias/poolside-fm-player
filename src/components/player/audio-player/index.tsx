/* eslint-disable jsx-a11y/media-has-caption */
// Libraries
import * as React from "react";
import ControlsMedia from "../controls-media/index";
import { IAudioPlayerProps, IAudioPlayerState, EPlayingStatus } from "../player.interfaces";

/**
 * @description Component Description
 * @author Jo√£o Dias
 * @date 2019-02-16
 * @returns {React.FunctionComponent<IAudioPlayerProps>}
 */
class AudioPlayer extends React.Component<IAudioPlayerProps, IAudioPlayerState> {
	private player: React.RefObject<HTMLAudioElement>;

	constructor(props: IAudioPlayerProps) {
		super(props);

		this.state = {
			player: null,
			status: EPlayingStatus.paused,
		};

		// Binds
		this.onClickOnPrevious = this.onClickOnPrevious.bind(this);
		this.onClickOnPlay = this.onClickOnPlay.bind(this);
		this.onClickOnNext = this.onClickOnNext.bind(this);

		// Refs
		this.player = React.createRef<HTMLAudioElement>();
	}

	/**
	 *
	 *
	 * @memberof AudioPlayer
	 */
	componentDidMount() {
		const { player } = this.state;
		if (this.player.current) {
			this.setState(
				{
					player: this.player.current,
				},
				() => {
					if (player) {
						player.currentTime = 0;
						player.pause();
					}
				},
			);
		}
	}

	/**
	 *
	 *
	 * @memberof AudioPlayer
	 */
	onClickOnPrevious() {
		console.log("previous");
	}

	/**
	 *
	 *
	 * @memberof AudioPlayer
	 */
	onClickOnNext() {
		console.log("next");
	}

	/**
	 *
	 *
	 * @memberof AudioPlayer
	 */
	onClickOnPlay() {
		const { status } = this.state;

		let nextStatus;

		switch (status) {
			default:
			case EPlayingStatus.paused:
				nextStatus = EPlayingStatus.playing;
				break;

			case EPlayingStatus.playing:
				nextStatus = EPlayingStatus.paused;
		}

		this.handleAudioPlayer(nextStatus);
	}

	/**
	 *
	 *
	 * @param {EPlayingStatus} status
	 * @memberof AudioPlayer
	 */
	handleAudioPlayer(status: EPlayingStatus) {
		const { player } = this.state;

		if (player) {
			switch (status) {
				case EPlayingStatus.playing:
					player.play();
					break;

				case EPlayingStatus.paused:
					player.pause();
					break;

				default:
					break;
			}

			this.updatePlayingStatus(status);
		}
	}

	/**
	 *
	 *
	 * @param {EPlayingStatus} status
	 * @memberof AudioPlayer
	 */
	updatePlayingStatus(status: EPlayingStatus) {
		this.setState({
			status,
		});
	}

	render() {
		const { status } = this.state;
		const { src } = this.props;

		return (
			<div id="media-player" className="media-player row">
				<audio
					ref={this.player}
					id="audio-player"
					src={src}
					data-testid="component-audio-player"
					autoPlay={false}
					className="sr-only"
				/>
				<ControlsMedia
					status={status}
					onClickOnPrevious={this.onClickOnPrevious}
					onClickOnPlay={this.onClickOnPlay}
					onClickOnNext={this.onClickOnNext}
				/>
			</div>
		);
	}
}

export default React.memo(AudioPlayer);
