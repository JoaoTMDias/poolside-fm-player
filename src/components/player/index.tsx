import ControlsMedia from "./controls-media/index";
import AudioPlayer from "./audio-player/index";

export { AudioPlayer, ControlsMedia };
