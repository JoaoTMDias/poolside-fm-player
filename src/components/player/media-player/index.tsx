/* eslint-disable jsx-a11y/media-has-caption */
// Libraries
import * as React from "react";
import { PlayerControllerContext } from "contexts/player-controller-context";
import CurrentSong from "../../current-song/index.component";
import ControlsMedia from "../controls-media/index";

/**
 * @description Component Description
 * @author Jo√£o Dias
 * @date 2019-02-16
 * @returns {React.FunctionComponent<IMediaPlayerProps>}
 */
const MediaPlayer = () => {
	return (
		<PlayerControllerContext.Consumer>
			{({
				artist,
				changeVolume,
				currentTime,
				next,
				previous,
				status,
				title,
				togglePlay,
				track,
			}) => {
				return (
					<div
						id="media-player"
						data-testid="component-media-player"
						className="media-player row"
						data-current-track={track?.current.toString()}
						data-playing-status={status}
					>
						<CurrentSong title={title} artist={artist} currentTime={`${currentTime}`} />
						<ControlsMedia
							status={status}
							onClickOnPrevious={previous}
							onTogglePlay={togglePlay}
							onClickOnNext={next}
							onChangeVolume={changeVolume}
						/>
					</div>
				);
			}}
		</PlayerControllerContext.Consumer>
	);
};

export default MediaPlayer;
