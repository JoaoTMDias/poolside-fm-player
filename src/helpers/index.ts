import hasPressedSpaceOrEnter from "./has-pressed-space-or-enter";

export * from "./logger.helper";
export * from "./key-codes";
export * from "./with-memo.helper";
export * from "./hold-on.helper";
export * from "./hover-focus.helper";
export * from "./history.helper";

export { hasPressedSpaceOrEnter };
