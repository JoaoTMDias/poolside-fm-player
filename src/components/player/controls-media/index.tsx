// Libraries
import * as React from "react";
import { Link } from "react-router-dom";
import { ROUTE_SETTINGS } from "data/constants/routes";
import { hasPressedSpaceOrEnter } from "helpers";
import * as S from "./controls-media.styled";
import { IControlsMediaProps, EPlayingStatus } from "../media-player/player.interfaces";
import ButtonVolume from "./buttons/button-volume";

/**
 * @description Controls - Media Buttons
 * @author Jo√£o Dias
 * @date 2019-02-16
 * @returns {React.FunctionComponent<IControlsMediaProps>}
 */
const ControlsMedia: React.FunctionComponent<IControlsMediaProps> = ({
	onClickOnPrevious,
	onTogglePlay,
	onClickOnNext,
	onClickOnVolume,
	status,
}) => {
	return (
		<S.Wrapper id="controls" className="controls">
			<S.Buttons id="controls-media" className="controls__media">
				<li className="controls-media__item controls-media__item--first">
					<button
						id="controls-media-button-previous"
						data-testid="component-controls-media-button-previous"
						title="Previous Song"
						aria-label="Click play the previous song"
						type="button"
						className="controls-media__button button button--large"
						onClick={() => {
							if (onClickOnPrevious) {
								onClickOnPrevious();
							}
						}}
						onKeyUp={(event: React.KeyboardEvent<HTMLButtonElement>) => {
							if (onClickOnPrevious && hasPressedSpaceOrEnter(event.keyCode)) {
								onClickOnPrevious();
							}
						}}
					>
						<svg
							className="icon"
							xmlns="http://www.w3.org/2000/svg"
							width="12"
							height="9"
							fill="none"
							viewBox="0 0 12 9"
						>
							<path
								fill="var(--color-icon, #000)"
								d="M12 0v9h-1V8h-1V7H9V6H8V5H7v4H6V8H5V7H4V6H3V5H2v4H0V0h2v4h1V3h1V2h1V1h1V0h1v4h1V3h1V2h1V1h1V0h1z"
							/>
						</svg>
					</button>
				</li>
				<li className="controls-media__item controls-media__item--middle">
					<button
						id="controls-media-button-play"
						data-testid="component-controls-media-button-play"
						title="Play song"
						aria-label="Click play the song"
						type="button"
						className="controls-media__button button button--large"
						onClick={() => {
							if (onTogglePlay) {
								onTogglePlay();
							}
						}}
						onKeyUp={(event: React.KeyboardEvent<HTMLButtonElement>) => {
							if (onTogglePlay && hasPressedSpaceOrEnter(event.keyCode)) {
								onTogglePlay();
							}
						}}
					>
						{status === EPlayingStatus.paused ? (
							<svg
								id="icon-pause"
								className="icon"
								xmlns="http://www.w3.org/2000/svg"
								width="9"
								height="9"
								fill="none"
								viewBox="0 0 9 9"
							>
								<path fill="var(--color-icon, #000)" d="M3 9V0h1v1h1v1h1v1h1v1h1v1H7v1H6v1H5v1H4v1H3z" />
							</svg>
						) : (
							<svg
								id="icon-play"
								className="icon"
								width="9"
								height="9"
								viewBox="0 0 9 9"
								fill="none"
								xmlns="http://www.w3.org/2000/svg"
							>
								<path d="M2 0H4V9H2V0Z" fill="var(--color-icon, #000)" />
								<path d="M5 0H7V9H5V0Z" fill="var(--color-icon, #000)" />
							</svg>
						)}
					</button>
				</li>
				<li className="controls-media__item controls-media__item--last">
					<button
						id="controls-media-button-next"
						data-testid="component-controls-media-button-next"
						title="Next Song"
						aria-label="Click play the next song"
						type="button"
						className="controls-media__button button button--large"
						onClick={() => {
							if (onClickOnNext) {
								onClickOnNext();
							}
						}}
						onKeyUp={(event: React.KeyboardEvent<HTMLButtonElement>) => {
							if (onClickOnNext && hasPressedSpaceOrEnter(event.keyCode)) {
								onClickOnNext();
							}
						}}
					>
						<svg
							className="icon"
							xmlns="http://www.w3.org/2000/svg"
							width="12"
							height="9"
							fill="none"
							viewBox="0 0 12 9"
						>
							<path
								fill="var(--color-icon, #000)"
								d="M0 9V0h1v1h1v1h1v1h1v1h1V0h1v1h1v1h1v1h1v1h1V0h2v9h-2V5H9v1H8v1H7v1H6v1H5V5H4v1H3v1H2v1H1v1H0z"
							/>
						</svg>
					</button>
				</li>
			</S.Buttons>
			<div className="controls-media__item">
				<ButtonVolume onClickOnVolume={onClickOnVolume} />
			</div>
			<div className="controls-media__item">
				<Link
					to={ROUTE_SETTINGS}
					id="controls-button-settings"
					data-testid="component-controls-button-settings"
					title="Settings"
					aria-label="Click to navigate to the app's settings page"
					className="controls-media__button button button--large"
				>
					<svg className="icon" xmlns="http://www.w3.org/2000/svg" width="9" height="9" fill="none" viewBox="0 0 9 9">
						<path
							fill="var(--color-icon, #000)"
							d="M8.413 3.913v-.587h-.587V2.152h.587v-.978h-.587V.587h-.978v.587H5.674V.587h-.587V0H3.913v.587h-.587v.587H2.152V.587h-.978v.587H.587v.978h.587v1.174H.587v.587H0v1.174h.587v.587h.587v1.174H.587v.978h.587v.587h.978v-.587h1.174v.587h.587V9h1.174v-.587h.587v-.587h1.174v.587h.978v-.587h.587v-.978h-.587V5.674h.587v-.587H9V3.913h-.587zM6.065 5.478h-.587v.587H3.522v-.587h-.587V3.522h.587v-.587h1.956v.587h.587v1.956z"
						/>
					</svg>
				</Link>
			</div>
		</S.Wrapper>
	);
};

ControlsMedia.defaultProps = {
	status: EPlayingStatus.paused,
};

export default React.memo(ControlsMedia);
