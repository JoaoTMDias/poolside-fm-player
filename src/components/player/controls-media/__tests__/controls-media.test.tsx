import React from "react";
import { shallow, mount } from "enzyme";
import ControlsMedia from "components/player/controls-media";
import { EPlayingStatus } from "components/player/media-player/player.interfaces";
import { KEY_CODES, hasPressedSpaceOrEnter } from "helpers";
import { HashRouter as Router } from "react-router-dom";
import ButtonVolume from "../button-volume";

describe("<ControlsMedia />", () => {
	it("should render the component without errors", () => {
		const component = shallow(<ControlsMedia status={EPlayingStatus.paused} />);

		expect(component).toMatchSnapshot();
	});

	describe("Play button", () => {
		it("should be paused by default", () => {
			const component = shallow(<ControlsMedia status={EPlayingStatus.paused} />);
			const playButton = component.find("[data-testid='component-controls-media-button-play']").first();
			const iconPlay = playButton.find("#icon-play");
			const iconPause = playButton.find("#icon-pause");

			expect(iconPlay.length).toBe(0);
			expect(iconPause.length).toBe(1);
		});

		it("should be play when a song is playing", () => {
			const component = shallow(<ControlsMedia status={EPlayingStatus.playing} />);
			const playButton = component.find("[data-testid='component-controls-media-button-play']").first();
			const iconPlay = playButton.find("#icon-play");
			const iconPause = playButton.find("#icon-pause");

			expect(iconPlay.length).toBe(1);
			expect(iconPause.length).toBe(0);
		});
	});

	describe("onTogglePlay", () => {
		it("should call on click", () => {
			const onTogglePlayMock = jest.fn();
			const component = shallow(<ControlsMedia onTogglePlay={onTogglePlayMock} status={EPlayingStatus.playing} />);
			const playButton = component.find("[data-testid='component-controls-media-button-play']").first();

			playButton.simulate("click");

			expect(onTogglePlayMock).toHaveBeenCalled();
		});

		it("should not call on click", () => {
			const onTogglePlayMock = jest.fn();
			const component = shallow(<ControlsMedia status={EPlayingStatus.playing} />);
			const playButton = component.find("[data-testid='component-controls-media-button-play']").first();

			playButton.simulate("click");

			expect(onTogglePlayMock).not.toHaveBeenCalled();
		});

		describe("should call on press", () => {
			it("on SPACE key", () => {
				const onTogglePlayMock = jest.fn();
				const component = shallow(<ControlsMedia onTogglePlay={onTogglePlayMock} status={EPlayingStatus.playing} />);
				const playButton = component.find("[data-testid='component-controls-media-button-play']").first();

				playButton.simulate("keyup", {
					keyCode: KEY_CODES.SPACE,
				});

				expect(onTogglePlayMock).toHaveBeenCalled();
			});

			it("on ENTER key", () => {
				const onTogglePlayMock = jest.fn();
				const component = shallow(<ControlsMedia onTogglePlay={onTogglePlayMock} status={EPlayingStatus.playing} />);
				const playButton = component.find("[data-testid='component-controls-media-button-play']").first();

				playButton.simulate("keyup", {
					keyCode: KEY_CODES.ENTER,
				});

				expect(onTogglePlayMock).toHaveBeenCalled();
			});
		});

		describe("should not call on press", () => {
			it("on SPACE key", () => {
				const onTogglePlayMock = jest.fn();
				const component = shallow(<ControlsMedia status={EPlayingStatus.playing} />);
				const playButton = component.find("[data-testid='component-controls-media-button-play']").first();

				playButton.simulate("keyup", {
					keyCode: KEY_CODES.SPACE,
				});

				expect(onTogglePlayMock).not.toHaveBeenCalled();
			});

			it("on ENTER key", () => {
				const onTogglePlayMock = jest.fn();
				const component = shallow(<ControlsMedia status={EPlayingStatus.playing} />);
				const playButton = component.find("[data-testid='component-controls-media-button-play']").first();

				playButton.simulate("keyup", {
					keyCode: KEY_CODES.ENTER,
				});

				expect(onTogglePlayMock).not.toHaveBeenCalled();
			});
		});
	});

	describe("hasPressedSpaceOrEnter", () => {
		it("should return false if keycode is not SPACE or ENTER", () => {
			expect(hasPressedSpaceOrEnter(KEY_CODES.ARROW_RIGHT)).toBe(false);
			expect(hasPressedSpaceOrEnter(KEY_CODES.ARROW_LEFT)).toBe(false);
			expect(hasPressedSpaceOrEnter(KEY_CODES.TAB)).toBe(false);
			expect(hasPressedSpaceOrEnter(KEY_CODES.ESC)).toBe(false);
		});

		it("should return true if keycode is SPACE or ENTER", () => {
			expect(hasPressedSpaceOrEnter(KEY_CODES.SPACE)).toBe(true);
			expect(hasPressedSpaceOrEnter(KEY_CODES.ENTER)).toBe(true);
		});
	});

	describe("onClickOnPrevious", () => {
		it("should call on click", () => {
			const onClickOnPreviousMock = jest.fn();
			const component = shallow(
				<ControlsMedia onClickOnPrevious={onClickOnPreviousMock} status={EPlayingStatus.playing} />,
			);
			const previousButton = component.find("[data-testid='component-controls-media-button-previous']").first();

			previousButton.simulate("click");

			expect(onClickOnPreviousMock).toHaveBeenCalled();
		});

		it("should not call on click if no props are passed", () => {
			const onClickOnPreviousMock = jest.fn();
			const component = shallow(<ControlsMedia status={EPlayingStatus.playing} />);
			const previousButton = component.find("[data-testid='component-controls-media-button-previous']").first();

			previousButton.simulate("click");

			expect(onClickOnPreviousMock).not.toHaveBeenCalled();
		});

		describe("should call on press", () => {
			it("on SPACE key", () => {
				const onClickOnPreviousMock = jest.fn();
				const component = shallow(
					<ControlsMedia onClickOnPrevious={onClickOnPreviousMock} status={EPlayingStatus.playing} />,
				);

				const previousButton = component.find("[data-testid='component-controls-media-button-previous']").first();

				previousButton.simulate("keyup", {
					keyCode: KEY_CODES.SPACE,
				});

				expect(onClickOnPreviousMock).toHaveBeenCalled();
			});

			it("on ENTER key", () => {
				const onClickOnPreviousMock = jest.fn();
				const component = shallow(
					<ControlsMedia onClickOnPrevious={onClickOnPreviousMock} status={EPlayingStatus.playing} />,
				);

				const previousButton = component.find("[data-testid='component-controls-media-button-previous']").first();

				previousButton.simulate("keyup", {
					keyCode: KEY_CODES.ENTER,
				});

				expect(onClickOnPreviousMock).toHaveBeenCalled();
			});
		});

		describe("should not call on press", () => {
			it("on SPACE key", () => {
				const onClickOnPreviousMock = jest.fn();
				const component = shallow(<ControlsMedia status={EPlayingStatus.playing} />);

				const previousButton = component.find("[data-testid='component-controls-media-button-previous']").first();

				previousButton.simulate("keyup", {
					keyCode: KEY_CODES.SPACE,
				});

				expect(onClickOnPreviousMock).not.toHaveBeenCalled();
			});

			it("on ENTER key", () => {
				const onClickOnPreviousMock = jest.fn();
				const component = shallow(<ControlsMedia status={EPlayingStatus.playing} />);

				const previousButton = component.find("[data-testid='component-controls-media-button-previous']").first();

				previousButton.simulate("keyup", {
					keyCode: KEY_CODES.ENTER,
				});

				expect(onClickOnPreviousMock).not.toHaveBeenCalled();
			});
		});
	});

	describe("onClickOnNext", () => {
		it("should call on click", () => {
			const onClickOnNextMock = jest.fn();
			const component = shallow(<ControlsMedia onClickOnNext={onClickOnNextMock} status={EPlayingStatus.playing} />);
			const nextButton = component.find("[data-testid='component-controls-media-button-next']").first();

			nextButton.simulate("click");

			expect(onClickOnNextMock).toHaveBeenCalled();
		});

		it("should not call on click", () => {
			const onClickOnNextMock = jest.fn();
			const component = shallow(<ControlsMedia status={EPlayingStatus.playing} />);
			const nextButton = component.find("[data-testid='component-controls-media-button-next']").first();

			nextButton.simulate("click");

			expect(onClickOnNextMock).not.toHaveBeenCalled();
		});

		describe("should call on press", () => {
			it("on SPACE key", () => {
				const onClickOnNextMock = jest.fn();
				const component = shallow(<ControlsMedia onClickOnNext={onClickOnNextMock} status={EPlayingStatus.playing} />);

				const nextButton = component.find("[data-testid='component-controls-media-button-next']").first();

				nextButton.simulate("keyup", {
					keyCode: KEY_CODES.SPACE,
				});

				expect(onClickOnNextMock).toHaveBeenCalled();
			});

			it("on ENTER key", () => {
				const onClickOnNextMock = jest.fn();
				const component = shallow(<ControlsMedia onClickOnNext={onClickOnNextMock} status={EPlayingStatus.playing} />);

				const nextButton = component.find("[data-testid='component-controls-media-button-next']").first();

				nextButton.simulate("keyup", {
					keyCode: KEY_CODES.ENTER,
				});

				expect(onClickOnNextMock).toHaveBeenCalled();
			});
		});

		describe("should not call on press", () => {
			it("on SPACE key", () => {
				const onClickOnNextMock = jest.fn();
				const component = shallow(<ControlsMedia status={EPlayingStatus.playing} />);

				const nextButton = component.find("[data-testid='component-controls-media-button-next']").first();

				nextButton.simulate("keyup", {
					keyCode: KEY_CODES.SPACE,
				});

				expect(onClickOnNextMock).not.toHaveBeenCalled();
			});

			it("on ENTER key", () => {
				const onClickOnNextMock = jest.fn();
				const component = shallow(<ControlsMedia status={EPlayingStatus.playing} />);

				const nextButton = component.find("[data-testid='component-controls-media-button-next']").first();

				nextButton.simulate("keyup", {
					keyCode: KEY_CODES.ENTER,
				});

				expect(onClickOnNextMock).not.toHaveBeenCalled();
			});
		});
	});

	describe("handleOnClick", () => {
		it("should call on click", () => {
			ButtonVolume.prototype.handleOnClick = jest.fn();

			const component = mount(
				<Router>
					<ControlsMedia status={EPlayingStatus.playing} />
				</Router>,
			);

			const ButtonComponent = component.find(ButtonVolume).first();
			const volumeButton = ButtonComponent.find("button[data-testid='component-controls-media-button-volume']").first();
			volumeButton.simulate("click");

			expect(ButtonVolume.prototype.handleOnClick).toHaveBeenCalled();
		});

		describe("should call on press", () => {
			it("on SPACE key", () => {
				ButtonVolume.prototype.handleOnClick = jest.fn();

				const component = mount(
					<Router>
						<ControlsMedia status={EPlayingStatus.playing} />
					</Router>,
				);

				const ButtonComponent = component.find(ButtonVolume).first();
				const volumeButton = ButtonComponent.find(
					"button[data-testid='component-controls-media-button-volume']",
				).first();

				volumeButton.simulate("keyup", {
					keyCode: KEY_CODES.SPACE,
				});

				expect(ButtonVolume.prototype.handleOnClick).toHaveBeenCalled();
			});

			it("on ENTER key", () => {
				ButtonVolume.prototype.handleOnClick = jest.fn();

				const component = mount(
					<Router>
						<ControlsMedia status={EPlayingStatus.playing} />
					</Router>,
				);

				const ButtonComponent = component.find(ButtonVolume).first();
				const volumeButton = ButtonComponent.find(
					"button[data-testid='component-controls-media-button-volume']",
				).first();

				volumeButton.simulate("keyup", {
					keyCode: KEY_CODES.ENTER,
				});

				expect(ButtonVolume.prototype.handleOnClick).toHaveBeenCalled();
			});
		});
	});

	describe("onChangeVolume", () => {
		it("should call on change on button input", () => {
			const onChangeVolumeMock = jest.fn();

			const component = mount(
				<Router>
					<ControlsMedia onChangeVolume={onChangeVolumeMock} status={EPlayingStatus.playing} />
				</Router>,
			);

			const ButtonComponent = component.find(ButtonVolume).first();
			ButtonComponent.instance().onChangeRangeInput("0.6");

			expect(onChangeVolumeMock).toHaveBeenCalled();
		});

		it("should not call on change on button input", () => {
			const onChangeVolumeMock = jest.fn();

			const component = mount(
				<Router>
					<ControlsMedia status={EPlayingStatus.playing} />
				</Router>,
			);

			const ButtonComponent = component.find(ButtonVolume).first();
			ButtonComponent.instance().onChangeRangeInput("0.6");

			expect(onChangeVolumeMock).not.toHaveBeenCalled();
		});
	});
});
