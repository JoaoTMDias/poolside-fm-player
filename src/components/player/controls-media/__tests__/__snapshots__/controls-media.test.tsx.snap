// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlsMedia /> should render the component without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-AxjAm ipxVdi controls"
        id="controls"
      >
        <ul
          class="sc-AxirZ ETQZp controls__media"
          id="controls-media"
        >
          <li
            class="controls-media__item controls-media__item--first"
          >
            <button
              aria-label="Click play the previous song"
              class="controls-media__button button button--large"
              data-testid="component-controls-media-button-previous"
              id="controls-media-button-previous"
              title="Previous Song"
              type="button"
            >
              <svg
                class="icon"
                fill="none"
                height="9"
                viewBox="0 0 12 9"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0v9h-1V8h-1V7H9V6H8V5H7v4H6V8H5V7H4V6H3V5H2v4H0V0h2v4h1V3h1V2h1V1h1V0h1v4h1V3h1V2h1V1h1V0h1z"
                  fill="var(--color-icon, #000)"
                />
              </svg>
            </button>
          </li>
          <li
            class="controls-media__item controls-media__item--middle"
          >
            <button
              aria-label="Click play the song"
              class="controls-media__button button button--large"
              data-testid="component-controls-media-button-play"
              id="controls-media-button-play"
              title="Play song"
              type="button"
            >
              <svg
                class="icon"
                fill="none"
                height="9"
                id="icon-pause"
                viewBox="0 0 9 9"
                width="9"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 9V0h1v1h1v1h1v1h1v1h1v1H7v1H6v1H5v1H4v1H3z"
                  fill="var(--color-icon, #000)"
                />
              </svg>
            </button>
          </li>
          <li
            class="controls-media__item controls-media__item--last"
          >
            <button
              aria-label="Click play the next song"
              class="controls-media__button button button--large"
              data-testid="component-controls-media-button-next"
              id="controls-media-button-next"
              title="Next Song"
              type="button"
            >
              <svg
                class="icon"
                fill="none"
                height="9"
                viewBox="0 0 12 9"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 9V0h1v1h1v1h1v1h1v1h1V0h1v1h1v1h1v1h1v1h1V0h2v9h-2V5H9v1H8v1H7v1H6v1H5V5H4v1H3v1H2v1H1v1H0z"
                  fill="var(--color-icon, #000)"
                />
              </svg>
            </button>
          </li>
        </ul>
        <div
          class="controls-media__item"
        >
          <button
            aria-label="Click to control the volume of the song"
            class="controls-media__button button button--large"
            data-testid="component-controls-media-button-volume"
            data-volume="1"
            id="controls-button-volume"
            title="Control Volume"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              height="9"
              viewBox="0 0 9 9"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.652.587V0H5.478v.587h-.587v.587h-.587v.587h-.587v.587h-1.37v.587h-.586v3.13h.587v.587h1.37v.587h.586v.587h.587v.587h.587V9h1.174v-.587h.587V.587h-.587z"
                fill="var(--color-icon, #000)"
              />
            </svg>
          </button>
        </div>
        <div
          class="controls-media__item"
        >
          <a
            aria-label="Click to navigate to the app's settings page"
            class="controls-media__button button button--large"
            data-testid="component-controls-button-settings"
            href="#/settings"
            id="controls-button-settings"
            title="Settings"
          >
            <svg
              class="icon"
              fill="none"
              height="9"
              viewBox="0 0 9 9"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.413 3.913v-.587h-.587V2.152h.587v-.978h-.587V.587h-.978v.587H5.674V.587h-.587V0H3.913v.587h-.587v.587H2.152V.587h-.978v.587H.587v.978h.587v1.174H.587v.587H0v1.174h.587v.587h.587v1.174H.587v.978h.587v.587h.978v-.587h1.174v.587h.587V9h1.174v-.587h.587v-.587h1.174v.587h.978v-.587h.587v-.978h-.587V5.674h.587v-.587H9V3.913h-.587zM6.065 5.478h-.587v.587H3.522v-.587h-.587V3.522h.587v-.587h1.956v.587h.587v1.956z"
                fill="var(--color-icon, #000)"
              />
            </svg>
          </a>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-AxjAm ipxVdi controls"
      id="controls"
    >
      <ul
        class="sc-AxirZ ETQZp controls__media"
        id="controls-media"
      >
        <li
          class="controls-media__item controls-media__item--first"
        >
          <button
            aria-label="Click play the previous song"
            class="controls-media__button button button--large"
            data-testid="component-controls-media-button-previous"
            id="controls-media-button-previous"
            title="Previous Song"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              height="9"
              viewBox="0 0 12 9"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0v9h-1V8h-1V7H9V6H8V5H7v4H6V8H5V7H4V6H3V5H2v4H0V0h2v4h1V3h1V2h1V1h1V0h1v4h1V3h1V2h1V1h1V0h1z"
                fill="var(--color-icon, #000)"
              />
            </svg>
          </button>
        </li>
        <li
          class="controls-media__item controls-media__item--middle"
        >
          <button
            aria-label="Click play the song"
            class="controls-media__button button button--large"
            data-testid="component-controls-media-button-play"
            id="controls-media-button-play"
            title="Play song"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              height="9"
              id="icon-pause"
              viewBox="0 0 9 9"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 9V0h1v1h1v1h1v1h1v1h1v1H7v1H6v1H5v1H4v1H3z"
                fill="var(--color-icon, #000)"
              />
            </svg>
          </button>
        </li>
        <li
          class="controls-media__item controls-media__item--last"
        >
          <button
            aria-label="Click play the next song"
            class="controls-media__button button button--large"
            data-testid="component-controls-media-button-next"
            id="controls-media-button-next"
            title="Next Song"
            type="button"
          >
            <svg
              class="icon"
              fill="none"
              height="9"
              viewBox="0 0 12 9"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 9V0h1v1h1v1h1v1h1v1h1V0h1v1h1v1h1v1h1v1h1V0h2v9h-2V5H9v1H8v1H7v1H6v1H5V5H4v1H3v1H2v1H1v1H0z"
                fill="var(--color-icon, #000)"
              />
            </svg>
          </button>
        </li>
      </ul>
      <div
        class="controls-media__item"
      >
        <button
          aria-label="Click to control the volume of the song"
          class="controls-media__button button button--large"
          data-testid="component-controls-media-button-volume"
          data-volume="1"
          id="controls-button-volume"
          title="Control Volume"
          type="button"
        >
          <svg
            class="icon"
            fill="none"
            height="9"
            viewBox="0 0 9 9"
            width="9"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.652.587V0H5.478v.587h-.587v.587h-.587v.587h-.587v.587h-1.37v.587h-.586v3.13h.587v.587h1.37v.587h.586v.587h.587v.587h.587V9h1.174v-.587h.587V.587h-.587z"
              fill="var(--color-icon, #000)"
            />
          </svg>
        </button>
      </div>
      <div
        class="controls-media__item"
      >
        <a
          aria-label="Click to navigate to the app's settings page"
          class="controls-media__button button button--large"
          data-testid="component-controls-button-settings"
          href="#/settings"
          id="controls-button-settings"
          title="Settings"
        >
          <svg
            class="icon"
            fill="none"
            height="9"
            viewBox="0 0 9 9"
            width="9"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.413 3.913v-.587h-.587V2.152h.587v-.978h-.587V.587h-.978v.587H5.674V.587h-.587V0H3.913v.587h-.587v.587H2.152V.587h-.978v.587H.587v.978h.587v1.174H.587v.587H0v1.174h.587v.587h.587v1.174H.587v.978h.587v.587h.978v-.587h1.174v.587h.587V9h1.174v-.587h.587v-.587h1.174v.587h.978v-.587h.587v-.978h-.587V5.674h.587v-.587H9V3.913h-.587zM6.065 5.478h-.587v.587H3.522v-.587h-.587V3.522h.587v-.587h1.956v.587h.587v1.956z"
              fill="var(--color-icon, #000)"
            />
          </svg>
        </a>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
